include/bluetooth/services/ots.h

```diff
@@ -468,6 +470,76 @@ struct bt_ots_feat {
 	uint32_t olcp;
 } __packed;
 
+/** @brief Object metadata request bit field values */
+enum {
+	/** @brief Request object name */
+	BT_OTS_METADATA_REQ_NAME                        = BIT(0),
+	/** @brief Request object type */
+	BT_OTS_METADATA_REQ_TYPE                        = BIT(1),
+	/** @brief Request object size */
+	BT_OTS_METADATA_REQ_SIZE                        = BIT(2),
+	/** @brief Request object first created time */
+	BT_OTS_METADATA_REQ_CREATED                     = BIT(3),
+	/** @brief Request object last modified time */
+	BT_OTS_METADATA_REQ_MODIFIED                    = BIT(4),
+	/** @brief Request object ID */
+	BT_OTS_METADATA_REQ_ID                          = BIT(5),
+	/** @brief Request object properties */
+	BT_OTS_METADATA_REQ_PROPS                       = BIT(6),
+	/** @brief Request all object metadata */
+	BT_OTS_METADATA_REQ_ALL                         = 0x7F,
+};
+
+/** @brief Date and Time structure */
+struct bt_ots_date_time {
+	uint16_t year;
+	uint8_t month;
+	uint8_t day;
+	uint8_t hours;
+	uint8_t minutes;
+	uint8_t seconds;
+};
+#define DATE_TIME_FIELD_SIZE 7
+
+/** @brief Metadata of an OTS object
+ *
+ * Used by the server as a descriptor for OTS object initialization.
+ * Used by the client to present object metadata to the application.
+ */
+struct bt_ots_obj_metadata {
+
+#if defined(CONFIG_BT_OTS)
+	/** @brief Object Name */
+	char                           *name;
+#endif /* CONFIG_BT_OTS */
+
+#if defined(CONFIG_BT_OTS_CLIENT)
+	/* TODO: Unify client/server name */
+	/** @brief Object name (client) */
+	char                           name_c[CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1];
+#endif /* CONFIG_BT_OTS_CLIENT */
```

Comment:
Could this be a union?

Reply:
It could, but I prefer not to.  It would be quite confusing, I think, to have two identically named variables where one is a pointer and the other a character array.  (I had some errors due to that while refactoring this.)
Propose to leave as is for now, plan is to be able to have them equal.